test/client/setup-test-framework.js

```diff
@@ -2,10 +2,23 @@
 /**
  * External dependencies
  */
-import { disableNetConnect } from 'nock';
+const nock = require( 'nock' );
 
-// It disables all network requests for all tests.
-disableNetConnect();
+// Disables all network requests for all tests.
+nock.disableNetConnect();
+
+beforeAll( () => {
+	// reactivate nock on test start
+	if ( ! nock.isActive() ) {
+		nock.activate();
+	}
+} );
+
+afterAll( () => {
+	// helps clean up nock after each test run and avoid memory leaks
+	nock.restore();
+	nock.cleanAll();
```

Comment:
I don't know `nock` but it looks legit. Have you tested how it impacts the performance of the test run? I remember having many issues with the `setup-test-framework.js` file and its impact on the total execution time.

Reply:
doesn't seem to have much impact. around 60s for a full run with a full cache both ways.